src/basicio.cpp

```diff
@@ -1037,8 +1038,9 @@ namespace Exiv2
 
     bool FileIo::eof() const
     {
-        assert(p_->fp_ != nullptr);
-        return feof(p_->fp_) != 0 || tell() >= (long)size();
+        if (p_->fp_ == nullptr)
+            return false;
```

Comment:
I am not sure if this is a good idea. I think `eof()` should in special cases return the value that will cause the API consumer to "take a step back" and "reconsider their decision to continue", which I think is rather `true` than `false`.

Or from a more pragmatic point of view: software should blow up in error cases ASAP, and saying that we reached `EOF` when the file is not opened would (at least I think) prevent further damage rather than saying `EOF` has not been reached.

Reply:
Right, I agree with your POV. I'll make the change and I'll check if something breaks with it.